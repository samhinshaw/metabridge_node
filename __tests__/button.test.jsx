import React from 'react';
import { shallow } from 'enzyme';
import Button from '../components/button';

test('Search renders correctly', () => {
  const component = shallow(
    <Button className="button is-large is-link" title="Tutorial" link="/help/tutorial" />
  );
  expect(component).toMatchSnapshot();
});
