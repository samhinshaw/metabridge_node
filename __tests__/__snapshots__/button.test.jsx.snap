// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    className="button is-large is-link"
    icon={null}
    link="/help/tutorial"
    title="Tutorial"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        href="/help/tutorial"
      >
        <a
          className="button is-large is-link"
        >
          <span>
            Tutorial
          </span>
        </a>
      </Link>,
      "className": "control",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <a
          className="button is-large is-link"
        >
          <span>
            Tutorial
          </span>
        </a>,
        "href": "/help/tutorial",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            <span>
              Tutorial
            </span>,
          ],
          "className": "button is-large is-link",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Tutorial",
            },
            "ref": null,
            "rendered": "Tutorial",
            "type": "span",
          },
        ],
        "type": "a",
      },
      "type": [Function],
    },
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          href="/help/tutorial"
        >
          <a
            className="button is-large is-link"
          >
            <span>
              Tutorial
            </span>
          </a>
        </Link>,
        "className": "control",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <a
            className="button is-large is-link"
          >
            <span>
              Tutorial
            </span>
          </a>,
          "href": "/help/tutorial",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              <span>
                Tutorial
              </span>,
            ],
            "className": "button is-large is-link",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Tutorial",
              },
              "ref": null,
              "rendered": "Tutorial",
              "type": "span",
            },
          ],
          "type": "a",
        },
        "type": [Function],
      },
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
